design/cluster-status-conditions.md

```diff
@@ -0,0 +1,39 @@
+# Cluster Status Condition
```

Comment:
This is more about rook-ceph operator status, so title and filename should make it clear.

Reply:
Worth mentioning that this work is heavily based on https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md